code/game/objects/items/bodybag.dm

```diff
@@ -8,8 +8,12 @@
 	var/unfoldedbag_path = /obj/structure/closet/body_bag
 
 /obj/item/bodybag/attack_self(mob/user)
-	deploy_bodybag(user, user.loc)
+	if(isopenturf(user.loc))
+		deploy_bodybag(user, user.loc)
+	else
+		to_chat(user, "<span class='warning'>You cannot deploy [src] inside of something!</span>")
```

Comment:
I don't think it needs any feedback

Reply:
It doesn't _need_ it per se.

But telling the user why they didn't accomplish the thing they tried is generally helpful and improves their experience.

Additionally, imagine the confusion of a new player who has deployed a body bag before, but suddenly cannot with no explanation. We are still meant to be a newbie friendly server.

Reply:
It's like trying to move into a wall, it doesn't need feedback.
Nothing happening to the player is enough feedback.

Reply:
> It's like trying to move into a wall.

It's really not. Especially since in most cases you can drop items or do other things(light a cigarette, send a PDA message, order stuff off your uplink, etc.). Moreover, the inability to move through walls is expected, because that's how walls are. This may not necessarily be so.